bunner-macroquad/src/bunner.rs

```diff
@@ -0,0 +1,194 @@
+use crate::{
+    child::Child, player_direction::PlayerDirection, player_state::PlayerState, position::Position,
+    resources::Resources, row::Row, splat::Splat, HEIGHT, WIDTH,
+};
+use macroquad::{
+    audio::play_sound_once,
+    prelude::{collections::storage, draw_texture, KeyCode, Texture2D, WHITE},
+};
+use std::collections::VecDeque;
+
+pub struct Bunner {
+    pub state: PlayerState,
+    pub timer: i32,
+    pub position: Position,
+    pub min_y: i32,
+    direction: PlayerDirection,
+    input_queue: VecDeque<PlayerDirection>,
+    image: Texture2D,
+}
+
+impl Bunner {
+    const MOVE_DISTANCE: i32 = 10;
+
+    pub fn new(position: Position) -> Self {
+        Self {
+            state: PlayerState::Alive,
+            timer: 0,
+            position,
+            min_y: position.y,
+            direction: PlayerDirection::Down,
+            input_queue: VecDeque::new(),
+            image: storage::get::<Resources>().blank_texture,
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        scroll_pos: i32,
+        rows: &mut [Box<dyn Row>],
+        input_queue: VecDeque<KeyCode>,
+    ) {
+        self.input_queue.append(
```

Comment:
There is a minor problem with the input.

At least on my system, if I hold a key for a little while, the game keeps repeating it for a certain amount of times, even after I've released they key (I guess the input is sampled and queued multiple times per frame).

It's unclear if it's intended or not. In some cases, allowing keeping the key down to repeat the movement is convenient (when the bunny needs to move many tiles), however, it ends being counterproductive with the current implementation, as it leads to loss of control.

In the original game, key holding does not repeat the movement - the bunny moves only tile, then waits for a new key up->down.

I think it's an improvement over the original game to make key holding repeat the movement, however, not with loss of control :grimacing: 
If you think that key holding should not be supported, then I think it's better not to repeat the movements at all, rather than causing loss of control

Reply:
The key repeat is not intentional. I think it’s caused by the frame cap code. If the macroquad frame is faster than the fps used by pygame, the input from the previous frame is repeated. Without this, the game would be too fast, and I didn’t want to introduce a frame time delta to the game so it would be closer to the original.